api/admin/controller/library_settings.py

```diff
@@ -294,19 +297,42 @@ def validate_geographic_areas(self, values):
                     elif len([x for x in us_search.query(state=state, returns=None) if x.county == city_or_county]):
                         locations["US"].append(value);
                     else:
-                        return UNKNOWN_LOCATION.detailed(_('Unable to locate "%(value)s".', value=value))
-
+                        if self.ask_registry(value):
+                            return self.ask_registry(value)
```

Comment:
This is going to call `ask_registry` twice, potentially making an unnecessary HTTP request. You can assign the result of `ask_registry` to a variable and then check it.